apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/TraceContext.java

```diff
@@ -427,7 +427,12 @@ public void asRootSpan(Sampler sampler) {
         transactionId.copyFrom(id);
         if (sampler.isSampled(traceId)) {
             flags = FLAG_RECORDED;
-            traceState.set(sampler.getSampleRate(), sampler.getTraceStateHeader());
+
+            if (Double.isNaN(traceState.getSampleRate())) {
```

Comment:
I would still do this check inside `TraceState#set()`. In fact, it already does it. Since you do this check here, you won't get the behaviour you want, e.g. throw an Exception when trying to override with a different value, or am I missing something?

Reply:
Yes, you are right, this extra check just avoids one possible buggy case by just attempting to ignore setting sample rate if it's already set. Thus removing it here (and only keep the one within `TraceState`) makes sense.